### PR TITLE
Occupation dropdown frontend

### LINKED ISSUE #7516: In the Patient registration page: Occupation dropdown is not suffecient.
**Is your feature request related to a problem? Please describe.**
In the Patient registration page: Occupation dropdown is not sufficient. it shows only Student, Businessman, Healthcare Worker, Healthcare lab Worker, Animal Handler and others!!!

**Describe the solution you'd like**
It should include all occupations.
I found this on GitHub, it might be helpful 

https://gist.github.com/ag14spirit/fbf877576c9d6b78899e3ad02fe92b50#file-occupation-dropdown-html

Atleast the below options are desirable.

- Healthcare Practitioners
- Paramedics
- Business, Executive, Management, and Financial Occupations
- Engineer
- Teacher 
- Other Professional Occupations:
- Office and Administrative Support Occupations:
- Chef or Head Cook
- Protective Service (e.g., Fire Fighting, Police Officer, Correctional Officer)
- Cook or Food Preparation Worker
- Food and Beverage Serving Worker (e.g., Bartender, Waiter, Waitress)
- Building and Grounds Cleaning and Maintenance
- Personal Care and Service (e.g., Hairdresser, Flight Attendant, Concierge)
- Sales Supervisor, Retail Sales
- Retail Sales Worker
- Insurance Sales Agent
- Sales Representative
- Real Estate Sales Agent
- Construction and Extraction (e.g., Construction Laborer, Electrician)
- Farming, Fishing, and Forestry
- Installation, Maintenance, and Repair
- Production Occupations
- Other Agriculture, Maintenance, Repair, and Skilled Crafts Occupation
- Aircraft Pilot or Flight Engineer
- Motor Vehicle Operator (e.g., Ambulance, Bus, Taxi, or Truck Driver)
- Other Transportation Occupation
- Military
- Homemaker
- Other Occupation
- Don't Know
- Not Applicable

### PR DESCRIPTION
**Pull request by hrit2773**
## Proposed Changes

- Fixes #7516 
- Changed the dropdown options
- Changed the form field component to auto complete form field

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

File: src/Components/Form/FormFields/Autocomplete.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ type AutocompleteFormFieldProps<T, V> = FormFieldBaseProps<V> & {
   error?: string;
 };
 
-const AutocompleteFormField = <T, V>(
+export const AutocompleteFormField = <T, V>(
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/Autocomplete.tsx]:
```suggestion
const AutocompleteFormField = <T, V>(
```

Why was this exported this way?

It is already exported as default.

File: src/Components/Patient/PatientRegister.tsx [hunk 2]
```diff
@@ -46,6 +46,7 @@ import { PatientModel, Occupation } from "./models";
 import PhoneNumberFormField from "../Form/FormFields/PhoneNumberFormField";
 import RadioFormField from "../Form/FormFields/RadioFormField";
 import { SelectFormField } from "../Form/FormFields/SelectFormField";
+import { AutocompleteFormField } from "../Form/FormFields/Autocomplete.js";
```
Comment (rithviknishad) [inline src/Components/Patient/PatientRegister.tsx]:
```suggestion
import AutocompleteFormField from "../Form/FormFields/Autocomplete.js";
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by hrit2773** [comment]
@rithviknishad yes yes I didn't notice that it was already exported...

**Review by rithviknishad — APPROVED** [review]

**Comment by hrit2773** [comment]
@rithviknishad I made the backend PR also check the link in the issues

**Comment by rithviknishad** [comment]
https://github.com/coronasafe/care/pull/2055#pullrequestreview-1981862720

**Comment by nihal467** [comment]
@hrit2773 we are having a cypress test failure, fix it

Navigate to the "patient_spec" folder, then locate the "patient_registration.cy.ts" file. In this file, update the Occupation selection component. Currently, it uses "cy.clickAndSelectOption". Change this to reuse the "cy.searchAndSelectOption" function as you change the occupation component from a select form field to an autocomplete form field.

**Comment by hrit2773** [comment]
https://github.com/coronasafe/care/pull/2055#pullrequestreview-1981862720 @rithviknishad @sainak can you review the backend PR also

**Comment by nihal467** [comment]
backend migration is failing

**Comment by rithviknishad** [comment]
@nihal467 the migration seems to be passing. Could you show the logs?

**Comment by hrit2773** [comment]
@rithviknishad maybe due to conflicting migrations after updating the branch the dependencies changed that's why maybe there was an issue anyways I deleted and re ran the migrations and pushed the code and that's why it seems to be passing. @nihal467 can you check again

**Review by nihal467 — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**